Classes/Outlet/AbstractOutlet.php

```diff
@@ -146,8 +176,97 @@ public function produce()
     {
         $data = $this->data;
         foreach ($this->pipesOut as $pipe) {
+            if ($pipe instanceof ViewAwarePipeInterface) {
+                $pipe->setView($this->view);
+            }
             $pipe->conduct($data);
         }
+
         return $data;
     }
+
+    /**
+     * @return ViewInterface
+     */
+    public function getView()
+    {
+        return $this->view;
+    }
+
+    /**
+     * @param ViewInterface $view
+     */
+    public function setView($view)
+    {
+        $this->view = $view;
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getArguments()
+    {
+        return $this->arguments;
+    }
+
+    /**
+     * @param array $arguments
+     */
+    public function setArguments($arguments)
+    {
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * @param OutletArgument $argument
+     */
+    public function addArgument(OutletArgument $argument)
+    {
+        $this->arguments[] = $argument;
+    }
+
+    /**
+     * Validate given $data based on configured argument validations
+     *
+     * @param $data
+     * @return Result
+     */
+    public function validate($data)
+    {
+        $this->validationResults = new Result();
+        foreach ($this->arguments as $argument) {
+            /** @var OutletArgument $argument */
```

Comment:
This annotation will not be needed if `$this->argument` gets annotated to type `OutletArgument[]`.